merino-cache/src/memory.rs

```diff
@@ -93,6 +143,13 @@ impl<S> Suggester<S> {
             ControlFlow::Continue(!should_remove)
         });
 
+        // remove any expired elements from the Pending table (There shouldn't be many.)
+        LOCK_TABLE.rcu(|table| {
+            let mut cleaned = HashMap::clone(table);
+            cleaned.retain(|_k, v| v > &mut start);
```

Comment:
```suggestion
            cleaned.retain(|_k, v| *v > start);
```

Reply:
Fixing this, but also moving it to it's own function in the `LOCK_TABLE` impl, because it probably should live there.